src/python/file-name-conventions.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+import yaml
+import subprocess
+import re
+import sys
+from pathlib import Path
+
+# Naming conventions
+NAMING_CONVENTIONS = {
+    'camelCase': r'^([a-z]+[a-zA-Z0-9]*)?\.',
+    'snake_case': r'^([a-z]+[a-z0-9_]*)?\.',
+    'kebab-case': r'^([a-z]+[a-z0-9\-]*)?\.',
+    'PascalCase': r'^([A-Z]+[a-zA-Z0-9]*)?\.',
+    'UPPER_CASE': r'^([A-Z]+[A-Z0-9_]*)?\.',
+    '*': r'^.*\.'
+}
+
+def load_config():
+    config_path = Path('cfg/file-name-conventions.yaml')
+    if config_path.exists():
+        with open(config_path, 'r') as file:
+            return yaml.safe_load(file)
+    return {}
+
+def check_file_naming(file_path, pattern):
+    return re.match(pattern, file_path.name)
+
+def main():
+    config = load_config()  # Load the configuration
+    default_case = config.get('default', 'snake_case')  # Get the default case
+    filetypes = config.get('filetypes', {})  # Get the file type specific conventions
+
+    # Validate the naming conventions against known conventions
+    all_conventions = set(filetypes.values()).union({default_case})
+    unrecognized_conventions = all_conventions - set(NAMING_CONVENTIONS.keys())
+    if unrecognized_conventions:
+        print('Error: Unrecognized naming convention in file-name-conventions.yaml')
+        print('Unrecognized naming conventions:', ', '.join(unrecognized_conventions))
+        sys.exit(1)
+
+    # Retrieve all tracked files
```

Comment:
Nit: use comment conventions from contributions guidelines

Reply:
Done, just removed most of them as they were redundant